pkg/graph/resolvers/key_contact.go

```diff
@@ -0,0 +1,320 @@
+		// Fetch the existing contact to check permissions and return after delete
+		existing, err := loaders.KeyContact.ByID.Load(ctx, id)
+		if err != nil {
+			logger.Warn("Failed to get contact with id", zap.Any("contactId", id), zap.Error(err))
+			return nil, nil
+		}
+
+		if existing == nil {
+			return nil, fmt.Errorf("contact with id %s not found", id)
+		}
+
+		// Check permissions
+		err = BaseStructPreDelete(logger, existing, principal, store, false)
+		if err != nil {
+			return nil, fmt.Errorf("error deleting contact. user doesn't have permissions. %s", err)
+		}
+
+		// Finally, delete the contact
+		returnedContact, err := storage.KeyContactDeleteContactByID(tx, principalAccount.ID, logger, id)
+
+		if err != nil {
+			return nil, fmt.Errorf("unable to delete contact. Err %w", err)
+		}
+		return returnedContact, nil
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return returnedContact, nil
+}
+
+// GetKeyContact retrieves a subject matter expert by its ID.
+// Returns the contact if found, or an error if not found or on failure.
+func GetKeyContact(ctx context.Context, logger *zap.Logger, principal authentication.Principal, store *storage.Store,
+	id uuid.UUID,
+) (*models.KeyContact, error) {
+	principalAccount := principal.Account()
+	if principalAccount == nil {
+		return nil, fmt.Errorf("principal doesn't have an account, username %s", principal.String())
+	}
+
+	contact, err := loaders.KeyContact.ByID.Load(ctx, id)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get contact with id %s: %w", id, err)
+	}
+
+	if contact == nil {
+		return nil, fmt.Errorf("contact with id %s not found", id)
+	}
+
+	return contact, nil
+}
+
+// GetKeyContacts retrieves all subject matter experts.
+// Returns the contacts if found, or an error if failure.
+func GetAllKeyContacts(ctx context.Context, logger *zap.Logger, store *storage.Store) ([]*models.KeyContact, error) {
+	contacts, err := loaders.KeyContact.GetAll.Load(ctx, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get all key contacts: %w", err)
+	}
+
+	if contacts == nil {
+		contacts = []*models.KeyContact{}
+	}
+
+	return contacts, nil
+}
+
+func sendKeyContactWelcomeEmail(
+	emailService oddmail.EmailService,
+	emailTemplateService email.TemplateService,
+	addressBook email.AddressBook,
+	contact *models.KeyContact,
+) error {
+	emailTemplate, err := emailTemplateService.GetEmailTemplate(email.KeyContactWelcomeTemplateName)
+	if err != nil {
+		return err
+	}
+
+	emailSubject, err := emailTemplate.GetExecutedSubject(email.NewKeyContactAddedSubjectContent(*contact))
+	if err != nil {
+		return err
+	}
+
+	emailBody, err := emailTemplate.GetExecutedBody(email.NewKeyContactAddedBodyContent(
+		emailService.GetConfig().GetClientAddress(),
+		*contact,
+	))
+	if err != nil {
+		return err
+	}
+
+	err = emailService.Send(
+		addressBook.DefaultSender,
+		[]string{contact.Email},
```

Comment:
If contact.Email is empty/nil (e.g., for mailbox contacts without user accounts), this might fail silently or send to wrong address. I recommend to add validation before sending

Reply:
Actually changing the field type to *String has made this changed to `[]string{*contact.Email}`, so does it still need to validate?